src/daft-micropartition/src/micropartition.rs

```diff
@@ -134,6 +134,28 @@ fn materialize_scan_task(
                 }) => {
                     let inference_options =
                         ParquetSchemaInferenceOptions::new(Some(*coerce_int96_timestamp_unit));
+
+                    let num_parallel_tasks = 8;
+
+                    let mut iceberg_delete_files: HashSet<&str> = HashSet::new();
```

Comment:
Nit: you could probably avoid the mut here with a `scan_task.sources.iter().flat_map()...` and I believe you might actually be able to `.collect()` directly into a HashSet as well